lib/mix/test/mix/tasks/deps_test.exs

```diff
@@ -308,6 +308,32 @@ defmodule Mix.Tasks.DepsTest do
     end)
   end
 
+  test "checks lock file has unused deps with --check-unused", context do
+    Mix.Project.push(DepsApp)
+
+    in_tmp(context.test, fn ->
+      Mix.Dep.Lock.write(%{whatever: "0.2.0", ok: "0.1.0"})
+      assert Mix.Dep.Lock.read() == %{whatever: "0.2.0", ok: "0.1.0"}
+      Mix.Tasks.Deps.Unlock.run(["--check-unused"])
+      assert Mix.Dep.Lock.read() == %{whatever: "0.2.0", ok: "0.1.0"}
+      error = "warning: unused dependencies in mix.lock file"
+      assert_received {:mix_shell, :error, [^error]}
+    end)
+  end
+
+  test "checks lock file has no unused deps with --check-unused", context do
```

Comment:
The tests can be merged into where you assert there is something unused, then you fix it, then nothing is unused.

Reply:
Done

Reply:
Unless you actually want me to call `Mix.Tasks.Deps.Unlock.run(["--unused"])` explicitly? That actually seems like what you meant now that I say it out loud.

Reply:
it looks great, thanks!

Reply:
This is now taken care of.